suites/src/main/java/com/hedera/block/suites/BaseSuite.java

```diff
@@ -121,6 +130,34 @@ protected static GenericContainer<?> createContainer() {
         return blockNodeContainer;
     }
 
+    /**
+     * Starts the block stream simulator in a separate thread.
+     *
+     * @param blockStreamSimulatorAppInstance the block stream simulator app instance
+     * @return a {@link Future} representing the asynchronous execution of the block stream simulator
+     */
+    protected Future<?> startSimulatorInThread(BlockStreamSimulatorApp blockStreamSimulatorAppInstance) {
+        return executorService.submit(() -> {
+            try {
+                blockStreamSimulatorAppInstance.start();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
```

Comment:
Using the thread pool like that will silence exceptions, we will not know in any way if an exception occurs in the submitted `Future`. If something is happening here, we will never know.

In order to not silence exceptions that can occur in the runnable, we need to override the `ThreadPoolExecutor#afterExecute(Runnable, Throwable)` method. In the javadoc of that method, there is an example override that shows how we can not silence exceptions, but here I provide also some very simple code to play around and test that (taken from the javadoc example):

```java
public class Main {
    public static void main(String[] args) {
        silencing();
//        nonSilencing();
    }

    private static void silencing() {
        final ExecutorService executorService = Executors.newFixedThreadPool(8);

        final Future<Object> future = executorService.submit(() -> {
            throw new RuntimeException("This is a test");
        });

        while (!future.isDone()) {
        }

        executorService.shutdownNow();
    }

    private static void nonSilencing() {
        final ExecutorService executorService = new ThreadPoolExecutor(8, 8, 0L, TimeUnit.MILLISECONDS,
            new LinkedBlockingQueue<>()) {
            @Override
            protected void afterExecute(final Runnable r, final Throwable t) {

                Throwable localThrowable = t;

                if (localThrowable == null && r instanceof Future<?> && ((Future<?>) r).isDone()) {
                    try {
                        final Object result = ((Future<?>) r).get();
                    } catch (CancellationException ce) {
                        localThrowable = ce;
                    } catch (ExecutionException ee) {
                        localThrowable = ee.getCause();
                    } catch (
                        InterruptedException ie) {// ignore/ reset
                        Thread.currentThread().interrupt();
                    }
                }
                if (localThrowable != null) {
                    localThrowable.printStackTrace(); // here you will see the exception printed, we handle as we wish
                }
            }
        };

        final Future<Object> future = executorService.submit(() -> {
            throw new RuntimeException("This is a test");
        });

        while (!future.isDone()) {
        }

        executorService.shutdownNow();
    }
}
```

I believe we should have a similar implementation, of course modified to fit our needs, otherwise we will not be able to catch problems that can occur which could lead to hours of investigation. This should be noted everywhere we use thread pools!

Reply:
Good point, I've added a custom thread executor, with simple funtionality to log the errors for now. We can extend as needed in the future..

Reply:
Agreed that for now we simply need to log, it is enough just to catch exceptions so we can debug if something is wrong inside these futures.